ext/phar/phar_object.c

```diff
@@ -2679,12 +2678,19 @@ PHP_METHOD(Phar, getPath)
  */
 PHP_METHOD(Phar, setAlias)
 {
-	char *alias, *error, *oldalias;
+	zend_string *new_alias = NULL;
+	char *error, *oldalias;
 	phar_archive_data *fd_ptr;
-	size_t alias_len, oldalias_len;
+	size_t oldalias_len;
 	int old_temp, readd = 0;
 
-	if (zend_parse_parameters(ZEND_NUM_ARGS(), "s", &alias, &alias_len) == FAILURE) {
+	// TODO Should this be using a "P" modifier?
+	if (zend_parse_parameters(ZEND_NUM_ARGS(), "S", &new_alias) == FAILURE) {
+		RETURN_THROWS();
+	}
+
+	if (ZSTR_LEN(new_alias) == 0) {
```

Comment:
Adding this check will make it impossible to unset an alias, i.e. you can now do `setAlias("")` to throw away the alias. I think that behaviour should be kept because there's no other way to do it otherwise.

Reply:
Right, I am confused how this is handled tho? As it feels like if you try to call `setAlias()` on two different `PharObject's you'll get an exception? But I didn't try it nor added tests, so I'm shunting this for later.

Reply:
> Right, I am confused how this is handled tho? As it feels like if you try to call setAlias() on two different `PharObject's you'll get an exception?

I don't understand the question

Reply:
From a quick glance it seems like trying to set an empty string as an alias for two different PharObjects would cause a PharException, but I might be missing something.